crates/nu-cli/src/repl.rs

```diff
@@ -709,6 +709,13 @@ pub fn eval_hook(
 ) -> Result<PipelineData, ShellError> {
     let value_span = value.span()?;
 
+    // Hooks can optionally be a record in this form:
+    // {
+    //     condition: {|before, after| ... }  # block that evaluates to true/false
+    //     code: # block or a string
+    // }
+    // The condition block will be run to check whether the main hook (in `code`) should be run.
+    // If it returns true (the default if a condition block is not specified), the hook should be run.
```

Comment:
Not really related to this change, I just had trouble understanding this part of the hooks code and decided to add comments. More deets in the original PR: https://github.com/nushell/nushell/pull/5982